canary/src/main/java/io/littlehorse/canary/kafka/KafkaTopicBootstrap.java

```diff
@@ -2,32 +2,35 @@
 
 import io.littlehorse.canary.Bootstrap;
 import io.littlehorse.canary.CanaryException;
+import io.littlehorse.canary.config.CanaryConfig;
 import io.littlehorse.canary.util.Shutdown;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.instrument.binder.kafka.KafkaClientMetrics;
 import java.util.List;
-import java.util.Map;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.kafka.clients.admin.AdminClient;
 import org.apache.kafka.clients.admin.KafkaAdminClient;
 import org.apache.kafka.clients.admin.NewTopic;
 import org.apache.kafka.common.errors.TopicExistsException;
 
 @Slf4j
-public class KafkaTopicBootstrap implements Bootstrap {
+public class KafkaTopicBootstrap extends Bootstrap implements MeterBinder {
 
-    public KafkaTopicBootstrap(
-            final String metricsTopicName,
-            final int topicPartitions,
-            final short topicReplicas,
-            final Map<String, Object> kafkaAdminConfigMap) {
+    private final AdminClient adminClient;
 
-        final AdminClient adminClient = KafkaAdminClient.create(kafkaAdminConfigMap);
-        Shutdown.addShutdownHook(adminClient);
+    public KafkaTopicBootstrap(final CanaryConfig config) {
+        super(config);
+
+        adminClient = KafkaAdminClient.create(config.toKafkaAdminConfig().toMap());
+        Shutdown.addShutdownHook("Topics Creator", adminClient);
 
         try {
-            final NewTopic canaryTopic = new NewTopic(metricsTopicName, topicPartitions, topicReplicas);
+            final NewTopic canaryTopic =
+                    new NewTopic(config.getTopicName(), config.getTopicPartitions(), config.getTopicReplicas());
 
             adminClient.createTopics(List.of(canaryTopic)).all().get();
```

Comment:
Suggestion:

Maybe it could be a good idea to add a timeout here

Reply:
done: `adminClient.createTopics(List.of(canaryTopic)).all().get(1, TimeUnit.SECONDS);`